### ISSUE TITLE
Cookie not being set in Safari on initial load

### ISSUE THREAD

**Issue by frubesss** [vercel/next.js#22974]
### What version of Next.js are you using?

10.0.8

### What version of Node.js are you using?

12.6.0

### What browser are you using?

Safari

### What operating system are you using?

macOS

### How are you deploying your application?

next dev

### Describe the Bug

When trying to set a cookie using document.cookie in a useEffect function the cookie is not being set on initial render when using the Safari browser. 

It is not until I refresh the page that the cookie is then set.

It works in all other browsers (Chrome, Firefox, Opera) on initial render.

### Expected Behavior

I would expect the cookie to be set on initial render of the application in the Safari browser.

### To Reproduce

1. Checkout this repository: https://github.com/frubesss/nextjs-safari-cookie-issue
2. Run `yarn install`
3. Run `yarn dev`
4. Navigate to `localhost:3000` in Safari
5. Check for the cookie `doSomethingOnlyOnce` being present

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
